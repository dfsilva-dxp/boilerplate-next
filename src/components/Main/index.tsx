const Main = () => (
  <main>
    <h1>Boilerplate Next</h1>
  </main>
);

export default Main;
