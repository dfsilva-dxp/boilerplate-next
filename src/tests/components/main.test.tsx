import { render, screen } from '@testing-library/react';

import Main from '@/components/Main';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    const heading = screen.getByRole('heading', { name: /boilerplate/i });
    const nextLogo = screen.getByRole('img', { name: /next.js logo/i });

    expect(heading).toBeInTheDocument();

    expect(nextLogo).toBeInTheDocument();
    expect(nextLogo).toHaveAttribute('src', '/next.svg');

    expect(container.firstChild).toMatchSnapshot();
  });
});
