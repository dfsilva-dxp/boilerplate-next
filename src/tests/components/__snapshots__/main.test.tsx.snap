// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-beyTiQ dgjEYL"
>
  <h1>
    Boilerplate
  </h1>
  <div>
    <img
      alt="Next.js Logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="37"
      src="/next.svg"
      style="color: transparent;"
      width="180"
    />
  </div>
</main>
`;
